.github/workflows/ci.yml

```diff
@@ -101,55 +101,51 @@ jobs:
-  #         credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
-  #         create_credentials_file: false
-  #     - name: "Setting up Python"
-  #       uses: actions/setup-python@v5.1.0
-  #       with:
-  #         python-version: "3.10" # Needs pybind >= 2.10.1 for Python >= 3.11
-  #     - name: "Installing Python packages"
-  #       run: |
-  #         python3 -m venv .venv
-  #         .venv/Scripts/activate.bat
-  #         python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
-  #     - name: "Installing requirements"
-  #       run: choco install ccache --yes
-  #     - name: "Configuring MSVC"
-  #       uses: ilammy/msvc-dev-cmd@v1.13.0
-  #     # Finally: build and run tests.
-  #     - name: "Building IREE"
-  #       env:
-  #         IREE_WRITE_REMOTE_CCACHE: ${{ needs.setup.outputs.write-caches }}
-  #         IREE_CCACHE_GCP_TOKEN: ${{ steps.gcp-auth.outputs.access_token }}
-  #         CCACHE_NAMESPACE: github-windows-2022
-  #       run: ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
-  #     - name: "Testing IREE"
-  #       run: ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
+  build_test_all_windows:
+    needs: setup
+    if: contains(fromJson(needs.setup.outputs.enabled-jobs), 'build_test_all_windows')
+    # TODO(saienduri): switch to self-hosted
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: bash
+    env:
+      BUILD_DIR: build-windows
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@v4.1.7
+        with:
+          submodules: true
+      - id: "gcp-auth"
+        name: "Authenticating to Google Cloud"
+        if: needs.setup.outputs.write-caches == 1
+        uses: "google-github-actions/auth@v1"
+        with:
+          token_format: "access_token"
+          credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: false
+      - name: "Setting up Python"
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.10" # Needs pybind >= 2.10.1 for Python >= 3.11
+      - name: "Installing Python packages"
+        run: |
+          python3 -m venv .venv
+          .venv/Scripts/activate.bat
+          python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
+      - name: "Installing requirements"
+        run: choco install ccache --yes
+      - name: "Configuring MSVC"
+        uses: ilammy/msvc-dev-cmd@v1.13.0
+      # Finally: build and run tests.
+      - name: "Building IREE"
+        env:
+          IREE_WRITE_REMOTE_CCACHE: ${{ needs.setup.outputs.write-caches }}
+          IREE_CCACHE_GCP_TOKEN: ${{ steps.gcp-auth.outputs.access_token }}
+          CCACHE_NAMESPACE: github-windows-2022
+        run: ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
+      - name: "Testing IREE"
+        run: ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
```

Comment:
Something is different between this and the package / release build.

This took 4h40m to build with an empty cache: https://github.com/iree-org/iree/actions/runs/9978560128/job/27575742025?pr=17930, while the Python compiler package takes ~3h40m: https://github.com/iree-org/iree/actions/runs/9972323508/job/27555349806.

The macos_x86_64 build takes ~3h hours (https://github.com/iree-org/iree/actions/runs/9963758090/job/27530935736) on in this workflow, which is about at the limit of acceptable for LLVM integrate PRs and other PRs that want to opt-in. 4h40m is too long IMO. Could still run this nightly though, or figure out how to get it down closer to the 3h40m release build timing.